jkube-kit/enricher/specific/src/main/java/org/eclipse/jkube/kit/enricher/specific/ServiceDiscoveryEnricher.java

```diff
@@ -0,0 +1,162 @@
+        listBuilder.accept(new TypedVisitor<ServiceBuilder>() {
+
+            @Override
+            public void visit(ServiceBuilder serviceBuilder) {
+                addAnnotations(serviceBuilder);
+            }
+
+        });
+    }
+
+    protected void addAnnotations(ServiceBuilder serviceBuilder) {
+        if (serviceBuilder.buildSpec() != null) {
+            List<ServicePort> ports = serviceBuilder.buildSpec().getPorts();
+            if (! ports.isEmpty()) {
+                ServicePort firstServicePort = ports.iterator().next();
+                port = firstServicePort.getPort().toString();
+                log.info("Using first mentioned service port '%s' " , port);
+            } else {
+                log.warn("No service port was found");
+            }
+        }
+
+        tryCamelDSLProject();
+
+        if (discoverable != null) {
+            String labelName = PREFIX;
+            String labelValue = getConfig(Config.discoverable, discoverable);
+            serviceBuilder.editOrNewMetadata().addToLabels(labelName, labelValue).and().buildMetadata();
+
+            log.info("Add %s label: \"%s\" : \"%s\"", PREFIX, labelName, labelValue);
+
+            Map<String, String> annotations = new HashMap<>();
+            annotations.put(PREFIX + "/" + DISCOVERY_VERSION, getConfig(Config.discoveryVersion, discoveryVersion));
+            annotations.put(PREFIX + "/" + SCHEME, getConfig(Config.scheme, scheme));
+
+            String resolvedPath = getConfig(Config.path, path);
+            if (resolvedPath != null) {
+                if (! resolvedPath.startsWith("/")) {
+                    resolvedPath = "/" + resolvedPath;
+                }
+                annotations.put(PREFIX + "/" + PATH, resolvedPath);
+            }
+            annotations.put(PREFIX + "/" + PORT, getConfig(Config.port, port));
+
+            String resolvedDescriptionPath = getConfig(Config.descriptionPath, descriptionPath);
+            if (resolvedDescriptionPath != null) {
+                if (! resolvedDescriptionPath.toLowerCase().startsWith("http") && ! resolvedDescriptionPath.startsWith("/")) {
+                    resolvedDescriptionPath = "/" + resolvedDescriptionPath;
+                }
+                annotations.put(PREFIX + "/" + DESCRIPTION_PATH , resolvedDescriptionPath);
+            }
+            for (String annotationName : annotations.keySet()) {
+                log.info("Add %s annotation: \"%s\" : \"%s\"", PREFIX, annotationName, annotations.get(annotationName));
+            }
+            serviceBuilder.editMetadata().addToAnnotations(annotations).and().buildMetadata();
+        }
+    }
+
+    public void tryCamelDSLProject(){
+        File camelContextXmlFile = new File(springConfigDir.getAbsoluteFile() + "/camel-context.xml");
+        if (camelContextXmlFile.exists()) {
+            try {
+                Document doc = DocumentBuilderFactory.newInstance().newDocumentBuilder().parse(camelContextXmlFile);
```

Comment:
This is reported as a vulnerability in Sonar (https://sonarcloud.io/project/issues?id=jkubeio_jkube&open=AXbM866k_aOP8HqgEQPC&pullRequest=509&resolved=false&types=VULNERABILITY):
![image](https://user-images.githubusercontent.com/488391/104309137-f7f72780-54d1-11eb-886b-e8a9cb275367.png)

Could you check if the following changes would still be compliant with the implementation:
```java
DocumentBuilderFactory df = DocumentBuilderFactory.newInstance();
df.setAttribute(XMLConstants.ACCESS_EXTERNAL_DTD, "");
df.setAttribute(XMLConstants.ACCESS_EXTERNAL_SCHEMA, "");
Document doc = df.newDocumentBuilder().parse(camelContextXmlFile)
```

Reply:
Done.